src/modm/platform/gpio/sam/module.lb

```diff
@@ -80,5 +107,10 @@ def build(env):
     bprops["gpios"] = driver["gpio"]
     env.template("config.hpp.in")
     env.copy("connector.hpp")
-    env.copy("pin.hpp")
+    if 'samg' in driver['type']:
+        env.copy("pin_samg.hpp", "pin.hpp")
+    else:
+        env.copy("pin.hpp")
```

Comment:
The two files have a lot in common, are you sure you want to duplicate it all? You could also select the family implementation within the file.

Reply:
No I'm not sure. It was easier to start that way and get it working, but it probably needs to be revisited.